tfjs-backend-webgpu/src/kernels/conv2d_mm_vec4_webgpu.ts

```diff
@@ -93,11 +93,57 @@ export class Conv2DMMVec4Program implements WebGPUProgram {
             getShapeCoords(this.convInfo.inShape)}) ? x[getFlatIndex(coord, ${
             getShapeCoords(
                 this.convInfo.inShape)}) / 4] : vec4(0.0, 0.0, 0.0, 0.0)`;
+    const sampleAWithRemainder = `int flatIndex = getFlatIndex(coord, ${
+        getShapeCoords(this.convInfo.inShape)});
+        int divBy4Remainder = flatIndex % 4;
+        int divBy4Index = flatIndex / 4;
+        vec4 curData = x[divBy4Index];
+        if (divBy4Remainder == 0)
+        {
+          temp = curData;
+        } else {
+          vec4 nextData = x[divBy4Index + 1];
```

Comment:
nonblocking: Could this end up being a redundant load with another one in the same shader invocation (for curData at another index, e.g. if work-per-thread > 1)? Perhaps there's an opportunity for optimization (though it might be tricky).

Reply:
Do you mean put nextData next to curData like below? If yes, done.
``` 
        vec4 curData = x[divBy4Index];
        vec4 nextData = x[divBy4Index + 1];

Reply:
What I mean is, could there be two calls to mm_readA in the same thread that result in loads like:
`nextData = x[4 + 1]`
`curData = x[5]`
? Deduplicating them would be hard (because the load is deep inside mm_readA), but probably saves some bandwidth.

> Do you mean put nextData next to curData like below? If yes, done.

I don't think this should be done. It could cause an unnecessary load in some cases (though the compiler may optimize to avoid that).

Reply:
Sorry for the misunderstanding. I thought your meaning was putting nextData and curData together would make the driver better to fetch data. Otherwise, in some case, the driver may hit unoptimized path since they are not adjacent in code even though nextData and curData are continuous in memory.

Yes, I agree that `curData` may be the same data with previous mm_readA's `nextData`. But it's hard to do the opt, which may make the code very complicated. I will leave a hint here for possible optimization.